### PR TITLE
Add Entity.getPriceMultiplier() method

### PR DESCRIPTION
**Pull request by RAldrich**
Refactor various Entity subtypes to move multiplier data from getCost() into getPriceMultiplier().

This enables external access to the multiplicative construction cost modifiers, which is needed to support
more accurate price calculations in MekHQ.

### CODE REVIEW

**Comment by RAldrich** [comment]
Manually tested, didn't notice any changes to costs (none are expected, this is just refactoring).

I have a MekHQ PR draft ready to go that's dependent on these changes.

Next step in this work would be to decouple the cost breakdown display from the cost calculations. Once that's done we'll have more freedom to redesign the cost calculations themselves, and maybe get Entity's cost structure into a form that plays nice with MekHQs concept of Parts.

File: megamek/src/megamek/common/Infantry.java [hunk 1]
```diff
@@ -1577,75 +1576,79 @@ else if(nSneak == 3) {
...
+    }
+
+    @Override
+    public double getPriceMultiplier() {
+        double priceMultiplier = 1.0;
 
         //Anti-Mek Trained Multiplier
         if (isAntiMekTrained()) {
-            multiplier = 1;
+            priceMultiplier *= 5.0;
```
Comment (sixlettervariables) [inline megamek/src/megamek/common/Infantry.java]: Why 5x for anti-mek trained? Previous code was 1x (unless this diff is odd).

Comment (RAldrich) [reply megamek/src/megamek/common/Infantry.java]: Oh sorry, I should have called that out in the commit. 5.0x is the correct multiplier per the Tech Manual (p. 282).

It looks like this flag might not actually be getting used though - I see no cost difference between the regular and Anti-Mech foot platoons with this change applied.

Comment (rroby1) [reply megamek/src/megamek/common/Infantry.java]: TM page 282 shows anti-mech training is x5 price multiplier

Comment (sixlettervariables) [reply megamek/src/megamek/common/Infantry.java]: @RAldrich that'll be an existing bug then.

**Pull request merged**